python/src/nanoarrow/schema.py

```diff
@@ -0,0 +1,448 @@
+    return Schema(Type.UINT16, nullable=nullable)
+
+
+def int32(nullable=True) -> Schema:
+    """Create an instance of a signed 32-bit integer type."""
+    return Schema(Type.INT32, nullable=nullable)
+
+
+def uint32(nullable=True) -> Schema:
+    """Create an instance of an unsigned 32-bit integer type."""
+    return Schema(Type.UINT32, nullable=nullable)
+
+
+def int64(nullable=True) -> Schema:
+    """Create an instance of a signed 32-bit integer type."""
+    return Schema(Type.INT64, nullable=nullable)
+
+
+def uint64(nullable=True) -> Schema:
+    """Create an instance of an unsigned 32-bit integer type."""
+    return Schema(Type.UINT64, nullable=nullable)
+
+
+def float16(nullable=True) -> Schema:
+    """Create an instance of a 16-bit floating-point type."""
+    return Schema(Type.HALF_FLOAT, nullable=nullable)
+
+
+def float32(nullable=True) -> Schema:
+    """Create an instance of a 32-bit floating-point type."""
+    return Schema(Type.FLOAT, nullable=nullable)
+
+
+def float64(nullable=True) -> Schema:
+    """Create an instance of a 64-bit floating-point type."""
+    return Schema(Type.DOUBLE, nullable=nullable)
+
+
+def string(nullable=True) -> Schema:
+    """Create an instance of a variable-length UTF-8 encoded string type."""
+    return Schema(Type.STRING, nullable=nullable)
+
+
+def large_string(nullable=True) -> Schema:
+    """Create an instance of a variable-length UTF-8 encoded string type
+    that uses 64-bit offsets."""
+    return Schema(Type.LARGE_STRING, nullable=nullable)
+
+
+def binary(byte_width=None, nullable=True) -> Schema:
+    """Create an instance of a variable or fixed-width binary type."""
+    if byte_width is not None:
+        return Schema(Type.FIXED_SIZE_BINARY, byte_width=byte_width, nullable=nullable)
+    else:
+        return Schema(Type.BINARY, nullable=nullable)
+
+
+def large_binary(nullable=True) -> Schema:
+    """Create an instance of a variable-length binary type that uses 64-bit offsets."""
+    return Schema(Type.LARGE_BINARY, nullable=nullable)
+
+
+def date32(nullable=True) -> Schema:
+    """Create an instance of a 32-bit date type (days since 1970-01-01)."""
+    return Schema(Type.DATE32, nullable=nullable)
+
+
+def date64(nullable=True) -> Schema:
+    """Create an instance of a 64-bit date type (milliseconds since 1970-01-01)."""
+    return Schema(Type.DATE64, nullable=nullable)
+
+
+def time32(unit, nullable=True) -> Schema:
+    """Create an instance of a 32-bit time of day type."""
+    return Schema(Type.TIME32, unit=unit, nullable=nullable)
+
+
+def time64(unit, nullable=True) -> Schema:
+    """Create an instance of a 64-bit time of day type."""
+    return Schema(Type.TIME64, unit=unit, nullable=nullable)
+
+
+def timestamp(unit, timezone=None, nullable=True) -> Schema:
+    """Create an instance of a timestamp type."""
+    return Schema(Type.TIMESTAMP, timezone=timezone, unit=unit, nullable=nullable)
```

Comment:
Idea for a follow-up PR, but we might want to do some mapping of "ms" -> TimeUnit.MILLI etc, so one can do `na.timestamp("ms")` ?